mysql-test/main/mysqld_option_err.test

```diff
@@ -46,6 +46,11 @@ mkdir $MYSQLTEST_VARDIR/tmp/mysqld_option_err;
 --error 7
 --exec $MYSQLD_BOOTSTRAP_CMD --skip-networking --datadir=$MYSQLTEST_VARDIR/tmp/mysqld_option_err --skip-grant-tables --plugin-dir=$MYSQLTEST_VARDIR/plugins --plugin-load=example=ha_example.so --plugin-example-enum-var=noexist >>$MYSQLTEST_VARDIR/tmp/mysqld_option_err/mysqltest.log 2>&1
 
+--echo Test to see if multiple invalid options will be displayed in the error output
+--replace_regex /^((?!nonexistent).)*$// /.*unknown/unknown/ 
+--error 2
+--exec $MYSQLD_BOOTSTRAP_CMD --skip-networking --datadir=$MYSQLTEST_VARDIR/tmp/mysqld_option_err --skip-grant-tables --nonexistentoption --alsononexistent --nonexistentvariable=1 2>&1
+
```

Comment:
what about `--sql-mode=invalid_value` ?

Reply:
what about ambiguous options, like `--character` ?

Reply:
generally, invalid arguments, like `--bootstrap=partstoob` or `--table-cache=enabled` ?

Reply:
Trying to start mariadbd with `--sql-mode=invalid_value --character --bootstrap=partstoob --table-cache=enabled`:
```
./build/sql/mariadbd --datadir=./data --user=mariadb --socket=/tmp/mysqld.sock --sql-mode=invalid_value --character --bootstrap=partstoob --table-open-cache=enabled
2023-12-18 16:22:20 0 [ERROR] ./build/sql/mariadbd: Error while setting value 'invalid_value' to 'sql_mode'
2023-12-18 16:22:20 0 [ERROR] ./build/sql/mariadbd: ambiguous option '--character' (character-set-client-handshake, character_sets_dir)
2023-12-18 16:22:20 0 [ERROR] ./build/sql/mariadbd: option '--bootstrap' cannot take an argument
2023-12-18 16:22:20 0 [Warning] option 'table_open_cache': unsigned value 0 adjusted to 10
```

I noticed that `--table-cache=enabled` does not prevent the server from starting up (even without my changes).

Reply:
One thing to point out (I mentioned it in the commit body and you also commented on this in [MDEV-26922](https://jira.mariadb.org/browse/MDEV-26922)) is that some options can be determined by the server as invalid "early" on and some options cannot be determined by the server as invalid until "later" on.

For example, ambiguous options like --character and invalid arguments like --bootstrap=partstoob can be determined as invalid by the server immediately.
On the other hand, the invalidity of unknown options like --nonexistentoption cannot be determined until other actions have taken place such as loading of all plugins.

So with these changes, it's still possible that the server does not print all invalid options. For example, `--nonexistent` is not printed here:
```
./build/sql/mariadbd --datadir=./data --user=mariadb --socket=/tmp/mysqld.sock --sql-mode=invalid_value --character --bootstrap=partstoob --nonexistentoption
2023-12-18 16:34:04 0 [ERROR] ./build/sql/mariadbd: Error while setting value 'invalid_value' to 'sql_mode'
2023-12-18 16:34:04 0 [ERROR] ./build/sql/mariadbd: ambiguous option '--character' (character-set-client-handshake, character_sets_dir)
2023-12-18 16:34:04 0 [ERROR] ./build/sql/mariadbd: option '--bootstrap' cannot take an argument
```

I went with this approach because it stays in line as much as possible with the current behavior. Also, if we want the server to display all invalid options at the same time, this means that sometimes the server will error out later than it could have.

Reply:
I can add another test case to `mysqld_option_err.test` to verify the behaviour on `--sql-mode=invalid_value --character --bootstrap=partstoob`.

Please let me know your thoughts on my above comment. Thanks!

Reply:
Yes, the test would be great, that was my point — to add a test for `--sql-mode=invalid_value --character --bootstrap=partstoob`. `handle_options()` returns an error in many different places, would be good to cover if not all of them, then at least most.

About "early" and "late" — I would think the current behavior is good enough. It's not a feature where a user needs 100% correctness, but just a helper to report as many errors as possible. You cannot report all errors correctly in all cases anyway — e.g. a missing value for an option will make my_getopt to swallow next option as a value and then everything will be out of sync, to resolve that you'd need to know user's intentions (like in `--aa --bb --cc` — are they options without values? or `--aa` with a value of `--bb` and `--cc` without? or `--aa` without a value and `--bb` with?). So, being "good enough", I think, should be the goal here.

Reply:
Agreed.

Test for `--sql-mode=invalid_value --character --bootstrap=partstoob` is added.